Patch-Source: https://github.com/gdm-settings/gdm-settings/pull/276
---
From 78d69d72e6f0f92929910b07a1372f8895be0e6b Mon Sep 17 00:00:00 2001
From: fossdd <fossdd@pwned.life>
Date: Wed, 4 Dec 2024 21:23:31 +0100
Subject: [PATCH] Use meson's debug option to detect development builds

meson's buildtype option contains not only if the build is not using
debug, but also configures the optimization option.

The issue was found in Alpine, where all meson packages are built with
'plain' as buildtype and not 'release', but both options set
debug=false.

Ref: https://mesonbuild.com/Builtin-options.html#details-for-buildtype
---
 data/icons/meson.build | 2 +-
 gdms/__init__.py       | 2 +-
 gdms/config.py.in      | 2 +-
 gdms/gui/window.py     | 4 ++--
 gdms/meson.build       | 2 +-
 5 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/data/icons/meson.build b/data/icons/meson.build
index b58a227a..434462c5 100644
--- a/data/icons/meson.build
+++ b/data/icons/meson.build
@@ -1,6 +1,6 @@
 hicolor_dir = get_option('datadir') / 'icons/hicolor'
 
-if get_option('buildtype') == 'release'
+if not get_option('debug')
   install_data('io.github.realmazharhussain.GdmSettings.svg',
     install_dir: hicolor_dir/'scalable/apps',
   )
diff --git a/gdms/__init__.py b/gdms/__init__.py
index bc11b869..765a6229 100644
--- a/gdms/__init__.py
+++ b/gdms/__init__.py
@@ -38,7 +38,7 @@
 PROJECT_NAME = 'gdm-settings'
 
 VERSION = config.version
-BUILD_TYPE = config.buildtype
+DEBUG = config.debug
 APP_DATA_DIR = APP_DIR + config.datadir + '/' + PROJECT_NAME
 
 
diff --git a/gdms/config.py.in b/gdms/config.py.in
index 67759fb2..683bb749 100644
--- a/gdms/config.py.in
+++ b/gdms/config.py.in
@@ -4,4 +4,4 @@ prefix = "@prefix@"
 version = '@version@'
 datadir = "@datadir@"
 localedir = "@localedir@"
-buildtype = '@buildtype@'
+debug = @debug@
diff --git a/gdms/gui/window.py b/gdms/gui/window.py
index 70863975..d3312ca0 100644
--- a/gdms/gui/window.py
+++ b/gdms/gui/window.py
@@ -5,7 +5,7 @@
 from gi.repository import Gtk
 from gi.repository import GObject
 
-from gdms import APP_NAME, APP_ID, BUILD_TYPE
+from gdms import APP_NAME, APP_ID, DEBUG
 from gdms.utils import BackgroundTask, GProperty, GSettings
 from gdms.cmd import Command
 from gdms.gresource import UbuntuGdmGresourceFile, BackgroundImageNotFoundError
@@ -58,7 +58,7 @@ class GdmSettingsWindow (Adw.ApplicationWindow):
     def __init__ (self, application, **props):
         super().__init__(**props)
 
-        if BUILD_TYPE != 'release':
+        if DEBUG:
             self.add_css_class('devel')
 
         self.application = application
diff --git a/gdms/meson.build b/gdms/meson.build
index 8f7f157a..a18a1949 100644
--- a/gdms/meson.build
+++ b/gdms/meson.build
@@ -10,7 +10,7 @@ configure_file(
     'version': meson.project_version(),
     'datadir': get_option('prefix') / get_option('datadir'),
     'localedir': get_option('prefix') / get_option('localedir'),
-    'buildtype': get_option('buildtype'),
+    'debug': get_option('debug'),
   }),
 )
 
